python/pyspark/pandas/groupby.py

```diff
@@ -2732,35 +2741,32 @@ def stat_function(col: Column) -> Column:
             return F.percentile_approx(col, 0.5, accuracy)
 
         return self._reduce_for_stat_function(
-            stat_function, only_numeric=numeric_only, bool_to_numeric=True
+            stat_function,
+            accepted_spark_types=(NumericType, BooleanType) if numeric_only else None,
+            bool_to_numeric=True,
         )
 
     def _reduce_for_stat_function(
         self,
         sfun: Callable[[Column], Column],
-        only_numeric: Optional[bool] = None,
-        bool_as_numeric: bool = False,
+        accepted_spark_types: Optional[Tuple[Type[DataType], ...]] = None,
         bool_to_numeric: bool = False,
     ) -> FrameLike:
         """Apply an aggregate function `sfun` per column and reduce to a FrameLike.
 
         Parameters
         ----------
         sfun : The aggregate function to apply per column
-        only_numeric: If True, only numeric columns are involved
-        bool_as_numeric: If True, boolean columns are seen as numeric columns (following pandas)
+        accepted_spark_types: Accepted spark types of columns to be aggregated;
+                              default None means all spark types are accepted
         bool_to_numeric: If True, boolean columns are converted to numeric columns
         """
         groupkey_names = [SPARK_INDEX_NAME_FORMAT(i) for i in range(len(self._groupkeys))]
         groupkey_scols = [s.alias(name) for s, name in zip(self._groupkeys_scols, groupkey_names)]
 
         agg_columns = []
         for psser in self._agg_columns:
-            if (
-                isinstance(psser.spark.data_type, NumericType)
-                or (bool_as_numeric and isinstance(psser.spark.data_type, BooleanType))
-                or not only_numeric
-            ):
+            if isinstance(psser.spark.data_type, accepted_spark_types):
```

Comment:
We should run this line when the `accepted_spark_types` is not None ?

Reply:
Nice catch!! Thanks.